src/main/java/com/google/codeu/servlets/AboutMeServlet.java

```diff
@@ -0,0 +1,68 @@
+package com.google.codeu.servlets;
+
+import java.io.IOException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.codeu.data.Datastore;
+import com.google.codeu.data.User;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+/**
+* Handles fetching and saving user data.
+*/
+@WebServlet("/about")
+public class AboutMeServlet extends HttpServlet {
+  private Datastore datastore;
+
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  /**
+   * Responds with the "about me" section for a particular user.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
+
+    String userEmail = request.getParameter("user");
+    if (userEmail == null || userEmail.equals("")) {
+      // Request is invalid, return empty response
+      return;
+    } 
+
+    User userData = datastore.getUser(userEmail);
+    if( userData == null || userData.getAboutMe() == null ) {
+      return;
+    }
+
+    response.getOutputStream().println(userData.getAboutMe());
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
```

Comment:
It checks for logged in status. But it does not check for if the logged in user is the same user that is being edited.
P.S: it was discussed in the project chat

Reply:
Which means I can edit someone else's about me

Reply:
Good point. I'm curious about how can one exploit it.

Reply:
Maybe you can append the parameterUsername in your post request. Then, check it on the servlet?

Reply:
Let's ask @huygaa11.

Reply:
Lemme test is on devserver

Reply:
test it*

Reply:
Nevermind it, the request is not going through. I think it's ok